logdna/logdna.py

```diff
@@ -50,32 +54,33 @@ def __init__(self, key, options={}):
         self.setLevel(logging.DEBUG)
         self.lock = threading.RLock()
 
-        self.session = requests.Session()
-        retry = Retry(connect=self.connection_retries, backoff_factor=self.retry_backoff_factor)
-        adapter = HTTPAdapter(max_retries=retry)
-        self.session.mount('https://', adapter)
-
     def buffer_log(self, message):
         if message and message['line']:
             if len(message['line']) > self.max_length:
                 message['line'] = message['line'][:self.max_length] + ' (cut off, too long...)'
                 if self.verbose in ['true', 'debug', 'd']:
                     logger.debug('Line was longer than %s chars and was truncated.', str(self.max_length))
 
-        self.buf_byte_length += sys.getsizeof(message)
+
 
         # Attempt to acquire lock to write to buf, otherwise write to secondary as flush occurs
         if not self.lock.acquire(blocking=False):
             self.secondary.append(message)
         else:
-            self.buf.append(message)
+            if self.buf_byte_length < self.buf_size_limit:
+                self.buf_byte_length += sys.getsizeof(message)
+                self.buf.append(message)
+            else:
+                logger.debug('The buffer size exceeded the limit: ', str(self.buf_size_limit))
```

Comment:
According to the [docs](https://docs.python.org/3/library/logging.html#logging.Logger.debug), I think, it's more correct if you specify `%s` or `%d`; i.e.
`logger.debug('The buffer size exceeded the limit: %s', str(self.buf_size_limit))`
or
`logger.debug('The buffer size exceeded the limit: %d', self.buf_size_limit)`

Reply:
Maybe we should use print out the internal logs to the terminal instead.